connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -369,9 +370,31 @@ else if (serverConnector != null && serverConnector.getHost() != null && serverC
         else if (serverConnector != null && serverConnector.getPort() > 0)
             builder.port(serverConnector.getPort());
 
-        log.info("Advertised URI: {}", builder.build());
+        URI uri = builder.build();
+        validateUriHost(uri);
+        log.info("Advertised URI: {}", uri);
 
-        return builder.build();
+        return uri;
+    }
+
+    /**
+     * Parses the uri and throws a more definitive error
+     * when the internal node to node communication can't happen due to an invalid host name.
+     */
+    private void validateUriHost(URI uri) {
+        if (uri.getHost() == null) {
+            String host = Utils.getHost(uri.getAuthority());
```

Comment:
Why use `Utils.getHost()` rather than `uri.getHost()`?

Reply:
I was wondering the same thing. Some local testing revealed that `Uri::getHost` returns `null` if the hostname is invalid, which also explains the null check at the line above.

Reply:
You are right, if the domain name is invalid then Uri parsing gives up and this is why authority will have the "host:port" pattern but the host and port will be null and thats why I am using  Utils.getHost uses regex to parse this out of authority(not reinventing the wheel as the method was already there). 

 The other alternative I had considered was "new URL(uri.toString()).getHost()" but in the future if java fixes the bug then it may also start returning null like URI.